File: LongestSubstringWithoutRepeatingCharacters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int a = 0, b = 0;
        int n = s.size();
        vector<char>v;
        vector<int> vi;
        int sum = 0;
        
        for(int i = 0 ; i < n ; i++)
        {
            int vs = v.size();
            for(int j = 0 ; j < vs ; j++)
            {
                if(s[i] == v[j])
                {
                    a = vi[j] + 1;
                    v.erase(v.begin(),v.begin() +j+1);
                    vi.erase(vi.begin(),vi.begin() +j+1);
  
                    break;
                }
            }
            
            v.push_back(s[i]);
            vi.push_back(i);
            sum = max(sum,b-a+1);
            b++;
        }
        
        return sum;
    }
};

File: LongestSubstringWithoutRepeatingCharacters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        unique = set()
        size = 0
        left = 0

        for right in range(len(s)):

            while left <= right and s[right] in unique:
                unique.remove(s[left])
                left += 1

            unique.add(s[right])
            size = max(size, right - left + 1)
            
        return size
